src/common/snippets/src/lowered/shape_inference/shape_infer_instances.cpp

```diff
@@ -0,0 +1,179 @@
+bool broadcast_merge_into(IShapeInferSnippets::VectorDims& dst,
+                          const IShapeInferSnippets::VectorDims& src,
+                          const ov::op::AutoBroadcastSpec& autob) {
+    auto broadcast_merge_dim = [](size_t& dst, const size_t& d1, const size_t& d2) {
+        if (d1 == d2 || d1 == 1 || d1 == IShapeInferSnippets::DYNAMIC_DIMENSION) {
+            dst = d2;
+        } else if (d2 == 1 || d2 == IShapeInferSnippets::DYNAMIC_DIMENSION) {
+            dst = d1;
+        } else {
+           return false;
+        }
+        return true;
+    };
+    // Ranks are both static.
+    const auto dst_rank = static_cast<int64_t>(dst.size());
+    const auto src_rank = static_cast<int64_t>(src.size());
+    switch (autob.m_type) {
+        case ov::op::AutoBroadcastType::NONE:
+            return true;
+        case ov::op::AutoBroadcastType::NUMPY: {
+            const auto new_rank = std::max(dst_rank, src_rank);
+            IShapeInferSnippets::VectorDims dims(new_rank);
+            bool success = true;
+            for (int64_t i = 0; i < new_rank; i++) {
+                auto dsti = i < (new_rank - dst_rank) ? 1 : dst[i - (new_rank - dst_rank)];
+                auto srci = i < (new_rank - src_rank) ? 1 : src[i - (new_rank - src_rank)];
+                success &= broadcast_merge_dim(dims[i], dsti, srci);
+            }
+            dst = std::move(dims);
+            return success;
+        }
+        case ov::op::AutoBroadcastType::PDPD: {
+            int64_t axis = autob.m_axis;
+            if (src_rank > dst_rank || axis < -1)
+                return false;
+
+            axis = (axis == -1) ? (dst_rank - src_rank) : axis;
+            if (src_rank + axis > dst_rank)
+                return false;
+
+            bool success = true;
+            for (int64_t i = 0; i < src_rank; ++i) {
+                if (dst[axis + i] != IShapeInferSnippets::DYNAMIC_DIMENSION &&
+                    src[i] != IShapeInferSnippets::DYNAMIC_DIMENSION) {
+                    if (src[i] > dst[axis + i])
+                        return false;
+                }
+                success &= broadcast_merge_dim(dst[axis + i], dst[axis + i], src[i]);
+            }
+            return success;
+        }
+        default:
+            OPENVINO_THROW("Unsupported auto broadcast type: ", autob.m_type);
+    }
+    return false;
+}
+bool merge_into(IShapeInferSnippets::VectorDims& dst, const IShapeInferSnippets::VectorDims& src) {
+    auto merge_dim = [](size_t& dst, const size_t& d1, const size_t& d2) {
+        if (d1 == d2 || d1 == IShapeInferSnippets::DYNAMIC_DIMENSION) {
+            dst = d2;
+        } else if (d2 == IShapeInferSnippets::DYNAMIC_DIMENSION) {
+            dst = d1;
+        } else {
+            return false;
+        }
+        return true;
+    };
+    if (dst.size() != src.size())
+        return false;
+
+    bool success = true;
+    for (size_t i = 0; i < dst.size(); i++)
```

Comment:
If we do so, then the following dimensions (after the unsuccessful one) won't be considered. Though there is no difference in the present case since we throw if `broadcast_*` returns false, there might be other cases in future when this behavior will be important.
The general Idea here is exactly to reproduce nGraph broadcasting behavior, but on VectorDims. This will make further implementations easier.